p/api/fever.php

```diff
@@ -0,0 +1,894 @@
+		foreach ($categories as $category) {
+			$groups[] = [
+				'id' => $category->id(),
+				'title' => $category->name()
+			];
+		}
+
+		return $groups;
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function getFavicons()
+	{
+		$favicons = array();
+
+		$dao = $this->getDaoForFeeds();
+		$myFeeds = $dao->listFeeds();
+
+		$salt = FreshRSS_Context::$system_conf->salt;
+
+		/** @var FreshRSS_Feed $feed */
+		foreach ($myFeeds as $feed) {
+
+			$id = hash('crc32b', $salt . $feed->url());
+			$filename = DATA_PATH . '/favicons/' . $id . '.ico';
+			if (!file_exists($filename)) {
+				continue;
+			}
+
+			$favicons[] = [
+				"id" => $feed->id(),
+				"data" => image_type_to_mime_type(exif_imagetype($filename)) . ";base64," . base64_encode(file_get_contents($filename))
+			];
+		}
+
+		return $favicons;
+	}
+
+	/**
+	 * @return int
+	 */
+	protected function getTotalItems()
+	{
+		$total_items = 0;
+
+		$dao = $this->getDaoForEntries();
+		$result = $dao->countFever();
+
+		if (!empty($result)) {
+			$total_items = $result['total'];
+		}
+
+		return $total_items;
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function getFeedsGroup()
+	{
+		$groups = array();
+		$ids = [];
+
+		$dao = $this->getDaoForFeeds();
+		$myFeeds = $dao->listFeeds();
+
+		/** @var FreshRSS_Feed $feed */
+		foreach ($myFeeds as $feed) {
+			$ids[$feed->category()][] = $feed->id();
+		}
+
+		foreach($ids as $category => $feedIds) {
+			$groups[] = [
+				'group_id' => $category,
+				'feed_ids' => implode(',', $feedIds)
+			];
+		}
+
+		return $groups;
+	}
+
+	/**
+	 * AFAIK there is no 'hot links' alternative in FreshRSS
+	 * @return array
+	 */
+	protected function getLinks()
+	{
+		return [];
+	}
+
+	/**
+	 * @param array $entries
+	 * @return string
+	 */
+	protected function entriesToIdList($entries = array())
+	{
+		$ids = [];
+		foreach ($entries as $id) {
+			$ids[] = (int) $id;
+		}
+
+		return implode(',', array_values($ids));
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function getUnreadItemIds()
+	{
+		$dao = $this->getDaoForEntries();
+		$entries = $dao->getUnread();
+		return $this->entriesToIdList($entries);
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function getSavedItemIds()
+	{
+		$dao = $this->getDaoForEntries();
+		$entries = $dao->getStarred();
+		return $this->entriesToIdList($entries);
+	}
+
+	protected function setItemAsRead($id)
+	{
+		$dao = $this->getDaoForEntries();
+		$dao->markRead($id, true);
+	}
+
+	protected function setItemAsUnread($id)
+	{
+		$dao = $this->getDaoForEntries();
+		$dao->markRead($id, false);
+	}
+
+	protected function setItemAsSaved($id)
+	{
+		$dao = $this->getDaoForEntries();
+		$dao->markFavorite($id, true);
+	}
+
+	protected function setItemAsUnsaved($id)
+	{
+		$dao = $this->getDaoForEntries();
+		$dao->markFavorite($id, false);
+	}
+
+	/**
+	 * TODO check this method for validity - is this required?
+	 * @param $html
+	 * @return string
+	 */
+	protected function rewriteUrls($html)
```

Comment:
FreshRSS already does some sanitizing, which might be enough. What is this function trying to do?

Reply:
Could you please give a try without this function, and see if it still works as intended with your client?

Reply:
Still works (as expected) without. These function were just small helper to convert a text url to a clickable link and some function to add a target="_blank" to links. I don't think it necessary.